lensless/mask.py

```diff
@@ -0,0 +1,321 @@
+import abc
+import progressbar
+import numpy as np
+import cv2 as cv
+from math import sqrt
+from perlin_numpy import generate_perlin_noise_2d
+from sympy.ntheory import quadratic_residues
+from scipy.signal import max_len_seq
+from waveprop.fresnel import fresnel_conv
+from waveprop.rs import angular_spectrum
+
+
+
+
+
+class Mask(abc.ABC):
+    """
+    Parent Mask class
+    """
+
+    def __init__(self, sensor_size_px: tuple, 
+                 sensor_size_m: tuple, 
+                 feature_size: tuple, 
+                 distance_sensor: float, 
+                 wavelength: float) -> None:
+        """
+        Parent mask contructor.
+        Attributes common to each type of mask.
+
+        Parameters
+        ----------
+        sensor_size_px: tuple (dim=2)
+            size of the sensor (px)
+        sensor_size_m: tuple (dim=2)
+            size of the sensor (m)
+        feature_size: float
+            size of the feature (m)
+        distance_sensor: float
+            distance between the mask and the sensor (m)
+        wavelength: float
+            wavelength (m)
```

Comment:
Let's say why it's used, i.e. to simulate the PSF. And note that we use 532nm as default since it's the mid-visible wavelength as they use in PhlatCam.

In future an RGB simulation would be better.